### ISSUE TITLE
Unable to install both aiohttp and Jinja2

### ISSUE THREAD

**Issue by lily-mara** [sdispater/poet#21]
I know this seems very strange, but Poet is not able to install aiohttp and Jinja2 if they are both in the same poetry.toml.

Basic config:

```toml
[package]
name = "poet-test"
version = "0.0.1"
description = "Poet test"

python = ["^3.6"]

authors = ["Nate Mara <natemara@gmail.com>"]

readme = "README.rst"

[dependencies]
aiohttp = "2.2.5"
Jinja2 = "2.9.6"
```

When running this with Python 3.6, pip 9.0.1, and poet 0.4.1, I get the following error:

```
λ poet install -vvv

Locking dependencies to poetry.lock

 / Resolving dependencies (1 sec, )


  [NotADirectoryError]
  [Errno 20] Not a directory: '/var/folders/hf/9sl9vzc515sbbnrfz15r4b6nz9mhjr/T/tmp839jvuj3build/changes/2143.bugfix'


Exception trace:
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/cleo/application.py in run() at line 95
   status_code = self.do_run(input_, output_)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/cleo/application.py in do_run() at line 188
   status_code = command.run(input_, output_)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/cleo/commands/command.py in run() at line 99
   return super(Command, self).run(i, o)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/console/commands/index_command.py in execute() at line 28
   super(IndexCommand, self).execute(i, o)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/console/commands/command.py in execute() at line 63
   return super(Command, self).execute(i, o)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/cleo/commands/command.py in execute() at line 108
   return self.handle()
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/console/commands/install.py in handle() at line 33
   installer.install(features=features, dev=dev)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/installer.py in install() at line 57
   self.lock(dev=dev)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/installer.py in lock() at line 262
   packages = self.resolve(deps)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/installer.py in resolve() at line 280
   return self._resolve(deps)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/poet/installer.py in _resolve() at line 320
   hashes = resolver.resolve_hashes(pinned)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/piptools/resolver.py in resolve_hashes() at line 77
   return {ireq: self.repository.get_hashes(ireq) for ireq in ireqs}
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/piptools/resolver.py in <dictcomp>() at line 77
   return {ireq: self.repository.get_hashes(ireq) for ireq in ireqs}
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/piptools/repositories/pypi.py in get_hashes() at line 169
   for candidate in matching_candidates
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/piptools/repositories/pypi.py in <setcomp>() at line 169
   for candidate in matching_candidates
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/piptools/repositories/pypi.py in _get_file_hash() at line 176
   download_dir=tmpdir, only_download=True, session=self.session
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/pip/download.py in unpack_url() at line 821
   hashes=hashes
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/pip/download.py in unpack_http_url() at line 663
   unpack_file(from_path, location, content_type, link)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/pip/utils/__init__.py in unpack_file() at line 605
   untar_file(filename, location)
 /users/nm46057/.anaconda3/envs/poet-test/lib/python3.6/site-packages/pip/utils/__init__.py in untar_file() at line 577
   with open(path, 'wb') as destfp:

install [-i|--index INDEX] [-f|--features FEATURES] [--no-dev] [--no-progress]
```

The really strange thing is that if I try installing each package individually, it works just fine, but if I have both of them in the file together it fails. Even if both packages are installed, it still fails.

**Comment by lily-mara** [comment]
This appears to be caused by a case insensitivity problem on OSX. Jinja2 has a file called `CHANGES` at the root of the project, and aiohttp has a directory called `changes` at the root of the project. Since poet installs everything in one go via pip, these names collide, and it causes a problem for Poet.

**Comment by lily-mara** [comment]
This is a problem with the `pip-tools` project, jazzband/pip-tools#557 is already tracking this issue.

**Comment by lily-mara** [comment]
Since the issue has been fixed in pip-tools 1.10.1, and this package uses ^1.9, I am closing this as fixed.
